packages/api/src/services/utils/uri-util.ts

```diff
@@ -118,6 +122,33 @@ export class UriUtil {
     return valid;
   }
 
+  /**
+   * Resolves a documentation URL based on the product version and endpoint path.
+   *
+   * This function constructs a complete documentation URL by combining the base documentation URL
+   * with the appropriate version and endpoint path. For unofficial versions (containing a hyphen)
+   * or when in development mode, it uses the latest unofficial version instead of the provided version.
+   *
+   * @param productVersion - The version of the product for which to retrieve documentation.
+   *                         If it contains a hyphen, it's considered an unofficial version.
+   * @param endpointPath - The specific documentation endpoint path to append to the URL.
+   * @returns A complete documentation URL string pointing to the specified resource.
+   * @throws {Error} If either productVersion or endpointPath is not provided.
+   *
+   * @example
+   * const docUrl = UriUtil.resolveDocumentationUrl('10.0.0', 'sparql-endpoint');
+   * // Returns: 'https://graphdb.ontotext.com/documentation/10.0.0/sparql-endpoint'
+   */
+  static resolveDocumentationUrl(productVersion: string, endpointPath: string): string {
+    if (!productVersion || !endpointPath) {
+      throw new Error('Product version and endpoint path are required for documentation URL resolution.');
+    }
+
+    const isUnofficialVersion = productVersion.includes('-');
```

Comment:
Unofficial versions might be formatted without hyphen, so IMO, a better approach would be to match with a regex validating that the product version is a string containing only a valid semantic versioning number.

Reply:
You may leave it as is, that's why I approved the MR